plugins/org.eclipse.elk.core/src/org/eclipse/elk/core/math/ElkMath.java

```diff
@@ -1236,4 +1237,20 @@ public static double shortestDistance(final ElkRectangle r1, final ElkRectangle
         // case 3
         return -Math.sqrt(vertDist * vertDist + horzDist * horzDist);
     }
+
+    /**
+    * Tests if the interior of this rect intersects the interior of a specified rect
+    */
+    public static boolean intersectsRect(ElkRectangle rect1, ElkRectangle rect2){
```

Comment:
Let's move this implementation directly into `ElkRectangle.intersects(...)`.

Reply:
This is done. Thanks.